### PR TITLE
test_network: fix tests for message_number/time_graph

### PR DESCRIPTION
**Pull request by kaushik-rohit**
1) fix the place where the patch was used
2) rather than mocking whole util module at once, patch
each function one by one, this provides us the flexibility to skip
mocking some particular functions, and also this is the correct way to
do mocking
3) TODO: some of the function on mocking produces the wrong result,
this has been ignored for now, but we need to investigate the cause.

Issue No.: https://github.com/prasadtalasila/IRCLogParser/issues/258

Changes proposed in this pull request:
- Described above

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@kaushik-rohit
@prasadtalasila

### CODE REVIEW

File: test/unit-test/test_lib/test_analysis/test_network.py [hunk 1]
```diff
@@ -25,48 +25,51 @@ def tearDown(self):
...
-        mock_util.create_connected_nick_list.return_value = util.load_from_disk(current_directory+ "/data/message_number_graph/conn_comp_list")
-        mock_util.check_if_msg_line.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/check_if_msg_line")
-        mock_util.correctLastCharCR.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/correctLastCharCR")
-        mock_util.get_nick_sen_rec.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/get_nick_sen_rec")
-        mock_util.extend_conversation_list.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/extend_conversation_list")
-        mock_util.get_nick_representative.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/get_nick_representative")
-        mock_util.correct_last_char_list.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/correct_last_char_list")
-        mock_util.get_year_month_day.side_effect = util.load_from_disk(current_directory + "/data/message_number_graph/get_year_month_day")
```
Comment (prasadtalasila) [inline test/unit-test/test_lib/test_analysis/test_network.py:47]: This advice on not mocking the whole modules must be documented in unit test patterns.

**Review by prasadtalasila — COMMENTED** [review]
@kaushik-rohit please update the testing patterns.
@bhuvangupta008 Can you deduce the reason for failure of the mocks in this specific case?

File: test/unit-test/test_lib/test_analysis/test_network.py (hunk 1 above)
Comment (kaushik-rohit) [reply test/unit-test/test_lib/test_analysis/test_network.py:47]: @prasadtalasila I have update the wiki.

**Comment by prasadtalasila** [comment]
@kaushik-rohit The test patterns page has side effects section that uses `mock_util.rec_list_splice.side_effect` as an example. This is the kind of anti-pattern you are advising against. Please update the side effects section of the wiki page as well.

**Pull request merged**